### PR TITLE
JIT ptrace

### PR DESCRIPTION
**Pull request by abocquet**
And adds a few random functions 

closes DYN-21

### CODE REVIEW

File: dynamiqs/rand.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    mat = jax.random.normal(key1, dims) + 1j * jax.random.normal(key2, dims)
+    return mat
+
+
+def hermitian(n: int, key=None) -> Array:
+    key = key if key is not None else jax.random.PRNGKey(0)
+    key1, key2 = jax.random.split(key, 2)
+    mat = 2 * jax.random.uniform(key1, (n, n)) - 1
+    mat = mat + 1j * (2 * jax.random.uniform(key2, (n, n)) - 1)
+    mat = 0.5 * (mat + dag(mat))
```
Comment (gautierronan) [inline dynamiqs/rand.py:19]:
```suggestion
    mat = complex_matrix((n, n), key=key)
    return 0.5 * (mat + dag(mat))
```
Why not something like this?

File: dynamiqs/rand.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def complex_matrix(dims: tuple[int, ...], key=None) -> Array:
+    key = key if key is not None else jax.random.PRNGKey(0)
+    key1, key2 = jax.random.split(key, 2)
+    mat = jax.random.normal(key1, dims) + 1j * jax.random.normal(key2, dims)
+    return mat
+
+
+def hermitian(n: int, key=None) -> Array:
```
Comment (gautierronan) [inline dynamiqs/rand.py:14]:
```suggestion
def hermitian_matrix(dims: tuple[int, ...], key=None) -> Array:
```
For consistency with `complex_matrix`.

Comment (gautierronan) [reply dynamiqs/rand.py:14]: Same comment for `dm` and `ket`.

**Review by gautierronan — CHANGES_REQUESTED** [review]
Nice ! Nit comments below.

File: dynamiqs/rand.py (hunk 1 above)
Comment (abocquet) [reply dynamiqs/rand.py:19]: Honestly I copy-pasted qutip's code, but this may be simpler indeed

File: dynamiqs/rand.py (hunk 2 above)
Comment (abocquet) [reply dynamiqs/rand.py:14]: It's just that `complex` is a reserved keywork of Python

Comment (gautierronan) [reply dynamiqs/rand.py:14]: My comment was also about the typing of the `dims` argument. You use `dims` in `complex_matrix`, and `n` in other functions.

**Review by gautierronan — APPROVED** [review]

**Pull request merged**